src/_pytest/mark/structures.py

```diff
@@ -283,28 +284,37 @@ def test_function():
     on the ``test_function`` object. """
 
     _config = None
+    _markers = set()
 
     def __getattr__(self, name):
         if name[0] == "_":
             raise AttributeError("Marker name must NOT start with underscore")
+
         if self._config is not None:
-            self._check(name)
+            self._update_markers(name)
+            if name not in self._markers:
+                warnings.warn(
+                    "Unknown pytest.mark.%s - is this a typo?  You can register "
+                    "custom marks to avoid this warning - for details, see "
+                    "https://docs.pytest.org/en/latest/mark.html" % name,
+                    UnknownMarkWarning,
+                )
+                if self._config.option.strict:
```

Comment:
Would you mind adding a `--strict-marks` alias on this PR, or would rather post pone it to another PR?

Reply:
Given the other problems here I'd rather push it to a third PR, just so we can ship this as soon as possible - I'd really like this out in time for PyCon which probably means next feature release.

Reply:
Sure, no worries. I've created https://github.com/pytest-dev/pytest/issues/5023 as a follow up. 👍